inference-engine/src/inference_engine/src/ie_core.cpp

```diff
@@ -62,18 +62,17 @@ Parsed<T> parseDeviceNameIntoConfig(const std::string& deviceName, const std::ma
     } else if (deviceName_.find("MULTI:") == 0) {
         deviceName_ = "MULTI";
         config_[InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES] = deviceName.substr(6);
-    } else if (deviceName_.find("AUTO") == 0) {
-        deviceName_ = "AUTO";
-        if (deviceName.size() > std::string("AUTO").size()) {
-            std::string deviceList = deviceName.substr(std::string("AUTO:").size());
-            if (deviceList.find("AUTO") != std::string::npos) {
-                IE_THROW() << "Device list for AUTO should not be AUTO";
-            }
-            config_[InferenceEngine::KEY_AUTO_DEVICE_LIST] = deviceName.substr(std::string("AUTO:").size());
+    } else if (deviceName.find("AUTO") == 0) {
+        deviceName_ = "MULTI";
+        if (deviceName.find("AUTO:") == 0) {
+            config_[InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES] =
+                deviceName.substr(std::string("AUTO:").size());
         }
+        // fixme: please correct me with a better key-value name
+        config_[InferenceEngine::PluginConfigParams::KEY_WORK_MODE] = InferenceEngine::PluginConfigParams::AUTO;
     } else {
-        if (deviceName_.empty()) {
-            deviceName_ = "AUTO";
+        if (deviceName_ == "AUTO") {
+            deviceName_ = "MULTI";
```

Comment:
do we need  here to set the config to the "AUTO" mode of the MULTI?config_[InferenceEngine::PluginConfigParams::KEY_WORK_MODE] = InferenceEngine::PluginConfigParams::AUTO;

Reply:
yes, thanks for pointing out this, will fix